File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False

        count={}
        for x in s:
            count[x] = count.get(x, 0) + 1

        for y in t:
            if y not in count:
                return False
            count[y] -= 1
            if count[y] == 0:
                del count[y]
        return len(count) == 0
        
        

File: 0242-valid-anagram/0242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    if (s.length !== t.length) return false;

    const count = {};

    for (let ch of s){
        count[ch] = (count[ch] || 0) + 1;
    }

    for (let ch of t){
        if (!count[ch]) return false;
        count[ch]--;
    }

    return true;
};